### PR TITLE
Add trace exports to Splunk HEC

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Adds trace exports support to Splunk HEC.

**Testing:**
Unit tests.

**Documentation:**
No doc - this is complementary to the metric export support with no additional config keys.

### CODE REVIEW

File: exporter/splunkhecexporter/exporter.go [hunk 1]
```diff
@@ -54,22 +62,23 @@ type exporterOptions struct {
 func New(
 	config *Config,
 	logger *zap.Logger,
-) (component.MetricsExporterOld, error) {
+) (TraceAndMetricExporter, error) {
 
 	if config == nil {
-		return nil, errors.New("nil config")
+		return splunkExporter{}, errors.New("nil config")
```
Comment (rmfitzpatrick) [inline exporter/splunkhecexporter/exporter.go:68]: These are unusable as is, correct?  Curious why this change is preferred.

Comment (atoulme) [reply exporter/splunkhecexporter/exporter.go:68]: Let me see if that change is actually needed. My go-fu is not that strong.

Comment (atoulme) [reply exporter/splunkhecexporter/exporter.go]: OK that change wasn't needed.

**Comment by flands** [comment]
@atoulme what is the latest on this?

**Comment by atoulme** [comment]
We had a first review, and I addressed it. It's ready for another review or merge.

**Comment by rmfitzpatrick** [comment]
> We had a first review, and I addressed it. It's ready for another review or merge.

I'm under the impression the diff coverage target is a requirement for acceptance so this will need additional testing.  Correct @tigrannajaryan?

File: exporter/splunkhecexporter/tracedata_to_splunk.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			numDroppedSpans++
+			continue
+		}
+		se := &splunkEvent{
+			Time:       timestampToEpochMilliseconds(span.StartTime),
+			Host:       host,
+			Source:     config.Source,
+			SourceType: config.SourceType,
+			Index:      config.Index,
+			Event:      span,
```
Comment (rmfitzpatrick) [inline exporter/splunkhecexporter/tracedata_to_splunk.go:55]: What assurances are there that this won't shift with unrelated internal type changes?  Seems less safe than the current metric translator: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/82b1adf3e12fdee23a2ea28c3c14b2c08a2ed477/exporter/splunkhecexporter/metricdata_to_splunk.go#L77

Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go:55]: I don't have any assurances, and I have superficial knowledge of HEC. I read the docs, used it, and it works reasonably well for me.

I would however point out that Splunk is in GA using this format, and the format has been used since the inception of HEC. If anything I would expect Splunk to change its HEC input in a backwards-compatible manner and with long deprecation periods.

What assurances do you require? How can I help?

Comment (rmfitzpatrick) [reply exporter/splunkhecexporter/tracedata_to_splunk.go:55]: Sry, I'm referring specifically to the json encoding of the `tracepb.Span`s as is and not the overall HEC event format, which I have not worked with before.  It's unclear to me that the spans are being marshaled as expected without using jsonpb/protojson directly.

**Comment by tigrannajaryan** [comment]
> I'm under the impression the diff coverage target is a requirement for acceptance so this will need additional testing. Correct @tigrannajaryan?

Yes, it would be nice if we could get a bit better coverage.

File: exporter/splunkhecexporter/tracedata_to_splunk.go (hunk 2 above)
Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go:55]: That's a good point. I'll add tests for that. Maybe that will help coverage enough too :)

**Comment by atoulme** [comment]
Ask and you shall receive - we have test, and better coverage 🎉

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by rmfitzpatrick** [comment]
Looks like the coverage checks weren't triggered with the latest push.  @atoulme maybe try pushing an amended commit?  LGTM otherwise (assuming the b64 ids are desired).

**Comment by jrcamp** [comment]
@atoulme try merging in master to pick up some changes that were made to codecov

**Comment by jrcamp** [comment]
I'm not sure why :/ It's working in other PRs. Can you try opening a new PR if it's not too much hassle and closing this one?

**Comment by atoulme** [comment]
Trying that.
